Python/Product/Analysis/LanguageServer/Server.WorkspaceSymbols.cs

```diff
@@ -96,11 +109,79 @@ private static IEnumerable<MemberResult> GetScopeVariables(ModuleAnalysis analys
             return res;
         }
 
+        private DocumentSymbol[] ToDocumentSymbols(List<MemberResult> members) {
+            var childMap = new Dictionary<MemberResult, List<MemberResult>>();
+            var topLevel = new List<MemberResult>();
+
+            foreach (var m in members) {
+                var parent = members.FirstOrDefault(x => x.Scope?.Node == m.Scope?.OuterScope?.Node && x.Name == m.Scope?.Name);
```

Comment:
We should be able to avoid this check by more carefully collecting values from scopes, yes? And save adding scope references to the MemberResult struct?

Reply:
Or arguably we can extract _nearly_ enough information from an AST walk (the names at least), and then use a direct analysis lookup to resolve those that we don't know for sure (like we do in the analysis classifier).

Reply:
Yes, I am actually going to write document symbols fetch using AST, but that is going to be September. https://github.com/Microsoft/vscode-python/issues/2430

Reply:
MemberResult is a public API, so adding/removing a field will be a breaking change.

Reply:
We'll need to revisit public surface going forward, add more interfaces since it gets harder to make changes as we go.

Reply:
Can make `MemberResult2` deriving from the base technically